### PR TITLE
Handle POST tracker web requests

### PR DESCRIPTION
**Pull request by DLance96**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+language: python
+python:
+    - "3.6"
+env:
+    - TOXENV=py36,coverage,mypy
+script:
+    - flake8 syncr_tracker tests
+    - pep8 syncr_tracker tests
```
Comment (matthewbentley) [inline .travis.yml]: s/pep8/pythoncodestyle

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+chardet==3.0.4
+coverage==4.5.1
+cryptography==2.1.4
+flake8==3.5.0
+identify==1.0.7
+idna==2.6
+mccabe==0.6.1
+Naked==0.1.31
+nodeenv==1.2.0
+pep8==1.7.1
```
Comment (matthewbentley) [inline requirements.txt]: rm this line

File: syncr_tracker/tracker.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    else:
+        # Return some sort of error
+        pass
+
+
+def handle_get(conn, request, addr):
+    pass
+
+
+def handle_post(conn, request):
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: what happens on bad requests? eg len == 33

File: syncr_tracker/tracker.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    Adds the pubkey to disk if it is a legal pairing
+    :param conn: TCP socket connection between server and client
+    :param request: [POST, node_id, pubkey]
+    :return:
+    """
+    if request[KEY_INDEX] == hashlib.sha256(request[VALUE_INDEX]
+                                            .encode('utf-8')).digest():
+        add_node_key_pairing(request)
+        print('Node/Key pairing added')
+        conn.send(bencode.encode('Node/Key pairing added'))
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: probably want a dict of `{"result": "OK"|"ERROR", "msg": "message"}`

Possibly create function that take `conn`, `result`, `message`, and does `conn.send`.

File: syncr_tracker/tracker.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+def request_post_drop_id(conn, request):
+    """
+    Adds node, ip, port tuples to appropriate drops in hashmap
+    :param conn: TCP socket connection between server and client
+    :param request: [POST, drop_id, [node_id, IP, port]]
+    :param addr: (IP, port)
+    :return:
+    """
+    if request[KEY_INDEX] in drop_availability.keys():
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: use defaultdict :) !
```
from collections import defaultdict
d = defaultdict(list)
d[key].append(value)
```
Will just work

File: syncr_tracker/tracker.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+            as pub_file:
+        pub_file.write(request[VALUE_INDEX].encode('utf-8'))
+
+
+def request_post_drop_id(conn, request):
+    """
+    Adds node, ip, port tuples to appropriate drops in hashmap
+    :param conn: TCP socket connection between server and client
+    :param request: [POST, drop_id, [node_id, IP, port]]
+    :param addr: (IP, port)
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: out of date

File: syncr_tracker/tracker.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+from constants import DROP_ID_BYTE_SIZE
+from constants import DROP_IP_INDEX
+from constants import DROP_NODE_INDEX
+from constants import DROP_PORT_INDEX
+from constants import KEY_INDEX
+from constants import NODE_ID_BYTE_SIZE
+from constants import TYPE_INDEX
+from constants import VALUE_INDEX
+
+drop_availability = dict()
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: `defaultdict(list)`

File: syncr_tracker/constants.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
+NODE_ID_BYTE_SIZE = 32
+DROP_ID_BYTE_SIZE = 64
+
+TYPE_INDEX = 0
+KEY_INDEX = 1
```
Comment (matthewbentley) [inline syncr_tracker/constants.py]: Call this `ID_INDEX`

File: tox.ini [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    coverage report --include 'tests*' --fail-under 100
+
+[testenv:mypy]
+setenv =
+    MYPYPATH = {toxinidir}
+mypy_paths =
+    syncr_tracker
+    tests
+commands =
+mypy {posargs:{[testenv:mypy]mypy_paths}}
```
Comment (matthewbentley) [inline tox.ini]: four spaces before mypy

File: syncr_tracker/tracker.py (hunk 3 above)
Comment (DLance96) [reply syncr_tracker/tracker.py]: 🐍

File: syncr_tracker/tracker.py [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+    send_server_response(conn, OK_RESULT, 'Drop availability updated')
+
+
+def generate_node_key_file_name(node_id):
+    """
+    Takes a node key provides where it's public key is stored
+    :param node_id:
+    :return: public key file
+    """
+    return 'pub_keys/{}.pub'.format(base64.b64encode(node_id, altchars=b'+-')
```
Comment (matthewbentley) [inline syncr_tracker/tracker.py]: Use `os.path.join`

**Pull request merged**